pkg/network/network.go

```diff
@@ -94,10 +103,16 @@ func NewConfigFromConfigMap(configMap *corev1.ConfigMap) (*Config, error) {
 	} else {
 		nc.IstioOutboundIPRanges = normalizedIpr
 	}
+
 	if ingressClass, ok := configMap.Data[DefaultClusterIngressClassKey]; !ok {
 		nc.DefaultClusterIngressClass = IstioIngressClassName
 	} else {
 		nc.DefaultClusterIngressClass = ingressClass
 	}
+
+	// If not in the config data then it'll be empty, which is ok, we'll
+	// just use the default value at that point
+	nc.DomainTemplate = configMap.Data[DomainTemplateKey]
```

Comment:
In this case I meant: Should we set `nc.DomainTemplate` to `DefaultDomainTemplate` here if the config map is empty?

Reply:
Sure I can do that. Just wondering though... what's the policy on defaults w.r.t. upgrades? Let's say the default value for a field is "foo" today, and the user never sets a value. Then they upgrade Knative and a new default value of "bar" has been defined. I can see an argument for existing resources sticking with 'foo' and I can see an argument for changing them to 'bar' since the user never specified a preference and it might be odd to have old and new resources behave differently w.r.t. a field that the user has never touched - thus a redeploy of the resource would result in a change in behavior.  Has Knative been following a pattern on this one?

@mattmoor this seems related to the API talks yesterday where you proposed exposing defaults as a way for people to learn about the more advanced config options.